pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -38,7 +39,13 @@ def __init__(self, min_steps: int, max_steps: int):
         super().__init__()
         self.min_steps: int = min_steps
 
-        if max_steps and max_steps < -1:
+        if max_steps is None:
+            rank_zero_deprecation(
+                "Setting `max_steps = None` is deprecated in v1.5 and will be removed in v1.7."
+                " Use `max_steps = -1` instead."
+            )
+            max_steps = -1
+        elif max_steps < -1:
```

Comment:
is the error message below wrong? what if `max_steps=0` ?

Reply:
`max_steps = 0` is fine and just results in training immediately ending

Reply:
what Ananth meant is that the formulation "positive integer or -1" is excluding the zero, but the 0 may be allowed. To be exact, we could say "must be a non-negative integer or -1".